testing-modules/testing-assertions/src/test/java/com/baeldung/optional/OptionalUnitTest.java

```diff
@@ -0,0 +1,69 @@
+package com.baeldung.optional;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class OptionalUnitTest {
+
+    private final String stringToTest = "testString";
+
+    @Test
+    public void givenOptionalWithValue_whenAssertThatIsNotEmpty_thenOk() {
+        assertThat(optionalWithValueMethodToTest()).isNotEmpty();
+    }
+
+    @Test
+    public void givenOptionalWithValue_whenAssertThatHasValue_thenOk() {
+        assertThat(optionalWithValueMethodToTest()).hasValue(stringToTest);
+    }
+
+    @Test
+    public void givenOptionalWithValue_whenAssertEqualsOptionalObject_thenOk() {
+        assertEquals(Optional.of(stringToTest), optionalWithValueMethodToTest());
+    }
+
+    @Test
+    public void givenOptionalWithValue_whenAssertEqualsGet_thenOk() {
+        Optional<String> optional = optionalWithValueMethodToTest();
+        assertEquals(stringToTest, optional.get());
+    }
+
+    @Test
+    public void givenOptionalWithValue_whenAssertTrueIsPresentAndGet_thenOk() {
+        Optional<String> optional = optionalWithValueMethodToTest();
+        assertTrue(optional.isPresent() && stringToTest.equals(optional.get()));
```

Comment:
This is a bad test - it's a bad bit of advice. If this fails, we don't know why.

Reply:
yes now that i see it i agree